lib/foreman_remote_execution_core/script_runner.rb

```diff
@@ -27,8 +27,8 @@ def start
 
       # pipe the output to tee while capturing the exit code in a file
       script = <<-SCRIPT
-          (#{su_prefix}#{remote_script} < /dev/null; echo $?>#{exit_code_path}) | /usr/bin/tee #{output_path}
```

Comment:
While looking at the csh (for me) unusual syntax, it seems the `/bin/sh -c` option is probably safest. To avoid the escaping, perhaps using the send-data way to transfer this executable script, the same way we actually send the data.